hydrosdk/deployment_configuration.py

```diff
@@ -0,0 +1,473 @@
+     If the key is empty, operator must be Exists; this combination means to match all values and all keys.
+    :param operator: Operator represents a key's relationship to the value. Valid operators are Exists and Equal. Defaults to Equal.
+     Exists is equivalent to wildcard for value, so that a pod can tolerate all taints of a particular category.
+    :param toleration_seconds: TolerationSeconds represents the period of time the toleration (which must be of effect NoExecute,
+     otherwise this field is ignored) tolerates the taint. By default, it is not set, which means tolerate the taint forever (do not evict).
+     Zero and negative values will be treated as 0 (evict immediately) by the system.
+    :param value: Value is the taint value the toleration matches to. If the operator is Exists, the value should be empty,
+     otherwise just a regular string.
+    """
+    operator: str
+    toleration_seconds: int = None
+    key: str = None
+    value: str = None
+    effect: str = None
+
+
+@enable_camel_case
+@dataclass
+class PodSpec:
+    """
+    PodSpec contains all configurations related to a pod.
+    """
+    node_selector: Dict[str, str] = field(default_factory=dict)
+    affinity: Affinity = None
+    tolerations: List[Toleration] = field(default_factory=list)
+
+
+@enable_camel_case
+@dataclass
+class DeploymentSpec:
+    """
+    DeploymentSpec contains all configurations related to a deployment.
+    """
+    replica_count: int = 1
+
+
+@enable_camel_case
+@dataclass
+class ResourceRequirements:
+    """
+    Resources required by the container.
+
+    :param limits: the maximum amount of compute resources allowed
+    :param requests: Requests describes the minimum amount of compute resources required.
+                     If Requests is omitted for a container, it defaults to Limits if that is explicitly specified,
+                     otherwise to an implementation-defined value.
+    """
+    limits: Dict[str, str]
+    requests: Dict[str, str] = None
+
+
+@enable_camel_case
+@dataclass
+class ContainerSpec:
+    """
+    ContainerSpec contains all configurations related to a container.
+    """
+    resources: ResourceRequirements = None
+
+
+@enable_camel_case
+@dataclass
+class DeploymentConfiguration:
+    """
+    DeploymentConfiguration encapsulates information about how your Servables and ModelVariants should run
+
+    :param name: Unique Name of a Deployment Configuration
+    :param hpa: HorizontalPodAutoScaler specification
+    :param pod: Pod specification
+    :param container: Container specification
+    :param deployment: Deployment specification
+    """
+    name: str
+    hpa: HorizontalPodAutoScalerSpec = None
+    pod: PodSpec = None
+    container: ContainerSpec = None
+    deployment: DeploymentSpec = None
+    _BASE_URL: str = "/api/v2/deployment_configuration"
+
+    @staticmethod
+    def list(cluster):
```

Comment:
For other entities we use list_all method. We should either rename this one or the others.

Reply:
I agree. What do you think, which one is better? Personally I am leaning towards `list` since `all` doesn't add any. meaning because `list` always returns all objects. We don't have `list_some` to have `list_all`.

Reply:
@KineticCookie @tidylobster @trthhrtz Do we have a consensus on this issue?

Reply:
I'm prefer to call a method just `list`